receiver/elasticapmreceiver/internal/mappers/intakeV2ToDerivedFields.go

```diff
@@ -89,3 +90,69 @@ func SetDerivedFieldsCommon(event *modelpb.APMEvent, attributes pcommon.Map) {
 		attributes.PutStr(elasticattr.EventOutcome, "unknown")
 	}
 }
+
+// Sets fields that are NOT part of OTel for errors. These fields are derived by the Enrichment lib in case of OTLP input
+func SetDerivedFieldsForError(event *modelpb.APMEvent, attributes pcommon.Map) {
+	attributes.PutStr(elasticattr.ProcessorEvent, "error")
+
+	if event.Error == nil {
+		return
+	}
+
+	if event.Error.Id != "" {
+		attributes.PutStr(elasticattr.ErrorID, event.Error.Id)
+	}
+	if event.Transaction != nil && event.Transaction.Id != "" {
+		attributes.PutStr(elasticattr.TransactionID, event.Transaction.Id)
+	}
+	if event.ParentId != "" {
+		attributes.PutStr(elasticattr.ParentID, event.ParentId)
+	}
+
+	if event.Error.Type != "" {
+		attributes.PutStr("error.type", event.Error.Type)
+	}
+	if event.Error.Message != "" {
+		attributes.PutStr("message", event.Error.Message)
+	}
+	attributes.PutStr(elasticattr.ErrorGroupingKey, event.Error.GroupingKey)
+	attributes.PutInt(elasticattr.TimestampUs, int64(event.Timestamp/1_000))
+
+	if event.Error.Culprit != "" {
+		attributes.PutStr("error.culprit", event.Error.Culprit)
+	}
+
+	if event.Error.Exception != nil {
+		if event.Error.Exception.Type != "" {
+			attributes.PutStr("exception.type", event.Error.Exception.Type)
+		}
+		if event.Error.Exception.Message != "" {
+			attributes.PutStr("exception.message", event.Error.Exception.Message)
+		}
+
+		if event.Error.Exception.Stacktrace != nil {
```

Comment:
We'll need to spend more time on this.

In the Elastic APM data model, the stacktrace is structured and at this point we only have the parsed stacktrace in the APM format. Kibana has a logic to show this in a native format of the given language. In OTel stacktrace is just a string.

I suggest we start with some meaningful default here for all stack traces regardless of the language. And then we can iterate on this in follow up PRs.

Reply:
> In OTel stacktrace is just a string.

Do you know if OTel defines any specific string format for stacktraces?

Reply:
Semconv seems to rely on languages SIG's: https://opentelemetry.io/docs/specs/semconv/attributes-registry/exception/

Reply:
Right - and the problem is, at this point we lost the language specific format, because it's already parsed in to the APM data format :(

Reply:
There's some logic in Kibana to re-create the stack trace in a language specific way. We could do a similar thing here. But I agree that we should start with something simple that works generically for all languages and iterate if needed.

Reply:
Opened https://github.com/elastic/opentelemetry-collector-components/issues/526 as a follow up.